corehq/apps/zapier/api/v0_5.py

```diff
@@ -87,8 +118,88 @@ def obj_get_list(self, bundle, **kwargs):
             ))
         return custom_fields
 
-    class Meta(CustomResourceMeta):
-        object_class = CustomField
+    class Meta(ZapierCustomFieldResource.Meta):
         resource_name = 'custom_fields'
-        include_resource_uri = False
-        paginator_class = DoesNothingPaginator
+
+"""
+class ZapierCustomActionFieldFormResource(ZapierCustomFieldResource):
+    required = fields.CharField(attribute='required', default='', null=True, blank=True)
+
+    def obj_get_list(self, bundle, **kwargs):
+        application_id = bundle.request.GET.get('application_id')
+        xmlns = bundle.request.GET.get('xmlns')
+        if not application_id or not xmlns:
+            return []
+
+        app = Application.get(application_id)
+        form = app.get_form_by_xmlns(xmlns)
+        custom_fields = []
+
+        for idx, question in enumerate(form.get_questions(app.langs)):
+            if self._has_default_label(question):
+                label = question['label'].split('/')[-1]
+            else:
+                label = question['label']
+            custom_fields.append(CustomActionField(
+                dict(
+                    type='unicode',
+                    key=self._build_key(question['hashtagValue']),
+                    label=label,
+                    required=json.dumps(question['required'])
+                )
+            ))
+
+        return custom_fields
+
+    class Meta(ZapierCustomFieldResource.Meta):
+        resource_name = 'custom_action_fields_form'
+"""
+
+#Map between keys and labels for general case properties (properties that every case has)
+CASE_PROPERTIES = {
+      "date_closed": "Date closed",
+      "date_modified": "Date modified",
+      "case_id": "Case ID",
+      "resource_uri": "Resource URI",
+      "user_id": "User ID",
+      "xform_ids": "XForm IDs",
+      "properties__case_name": "Case name",
+      "properties__case_type": "Case type",
+      "properties__owner_id": "Owner ID",
+      "properties__date_opened": "Date opened",
+      "properties__external_id": "External ID",
+}
+
+
+class ZapierCustomFieldCaseResource(ZapierCustomFieldResource):
+    """
+    Generates custom trigger field labels for cases
+    """
+
+    def obj_get_list(self, bundle, **kwargs):
+        custom_fields = []
+        domain = bundle.request.GET.get('domain')
+        case_type = bundle.request.GET.get('case_type')
+
+        for prop in get_case_properties_for_case_type(domain, case_type):
+            custom_fields.append(CustomField(
+                dict(
+                    type='unicode',
+                    key="properties__" + prop,
+                    label=prop
```

Comment:
The properties you have in the `CASE_PROPERTIES` list all have nice formatting without underscores. I think this will show up in zapier with underscores as the names. Do you think it would be better to remove the underscores in these properties (see [replace](https://docs.python.org/2/library/string.html#string.replace)), or do you want to add underscores to the properties above? I think you should be consistent either way.